MsCorePkg/MacAddressEmulationDxe/MacAddressEmulationDxe.h

```diff
@@ -0,0 +1,73 @@
+/** @file
+
+  DXE Driver header for handling MAC Address Emulation.
+
+  Copyright (C) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#pragma once
+
+#include <Uefi.h>
+#include <Protocol/SimpleNetwork.h>
+
+//
+// Utility macro that splits EFI_MAC_ADDRESS into
+// a set of CHAR8 arguments (e.g. for printing)
+//
+#define MAC_TO_CHAR8_ARGS(MacAddress) \
+  ((UINT8 *)&MacAddress)[0], \
+  ((UINT8 *)&MacAddress)[1], \
+  ((UINT8 *)&MacAddress)[2], \
+  ((UINT8 *)&MacAddress)[3], \
+  ((UINT8 *)&MacAddress)[4], \
+  ((UINT8 *)&MacAddress)[5]
+
+typedef struct {
+  VOID            *Registration;
+  BOOLEAN         Assigned;
+  EFI_MAC_ADDRESS EmulationAddress;
+  EFI_MAC_ADDRESS PermanentAddress;
+} MAC_EMULATION_SNP_NOTIFY_CONTEXT;
+
+typedef
+BOOLEAN
+(*SNP_MATCH_FUNCTION)(
+  IN CONST EFI_HANDLE SnpHandle,
+  IN CONST EFI_SIMPLE_NETWORK_PROTOCOL *Snp,
+  OPTIONAL IN CONST VOID *SnpContext
+  );
+
+BOOLEAN
+SnpSupportsMacEmuCheck (
```

Comment:
Prefer not to comment on test functions since those usually take the form of

```
/**
  Unit test for ValidateBlobWithXdrCertificates () API with all failed validation.

  @param[in]  Context    [Optional] An optional parameter that enables:
                         1) test-case reuse with varied parameters and
                         2) test-case re-entry for Target tests that need a
                         reboot.  This parameter is a VOID* and it is the
                         responsibility of the test author to ensure that the
                         contents are well understood by all test cases that may
                         consume it.

  @retval  UNIT_TEST_PASSED             The Unit test has completed and the test
                                        case was successful.
  @retval  UNIT_TEST_ERROR_TEST_FAILED  A test case assertion has failed.
**/
```

which will be the same for every single test case.